Devscord.DiscordFramework/Middlewares/Factories/MessageContextFactory.cs

```diff
@@ -0,0 +1,23 @@
+﻿using Autofac;
+using Devscord.DiscordFramework.Framework.Commands.Parsing;
+using Devscord.DiscordFramework.Middlewares.Contexts;
+using System;
+
+namespace Devscord.DiscordFramework.Middlewares.Factories
+{
+    class MessageContextFactory
+    {
+        private CommandParser _commandParser;
+
+        public MessageContextFactory(IComponentContext context)
+        {
+            this._commandParser = context.Resolve<CommandParser>();
+        }
+
+        public MessageContext Create(DateTime sendAt, string message)
+        {
+            bool IsBotCommand() => this._commandParser.Parse(message, sendAt).IsCommandForBot;
```

Comment:
dla upewnienia się - to na pewno wykonuje się dopiero podczas użycia IsBotCommand?
nie powinno być => () => this._commandParser.Parse(message, sendAt).IsCommandForBot;
(tylko się pytam, bo nie jestem pewien)

Reply:
ja bym tu po prostu brał wynik

lazy loading tej informacji ma sens?
to znaczy
parsowanie jest kosztowne - wiadomo
ale czy przypadkiem nie jest tak, że parsujemy każdą wiadomość?